src/Common/SLRUCachePolicy.h

```diff
@@ -326,16 +326,16 @@ class SLRUCachePolicy : public ICachePolicy<Key, Mapped, HashFunction, WeightFun
             else
             {
                 current_weight_lost += cell.size;
+                if (!is_protected)
```

Comment:
We already checked that `entry.is_protected == false` in this scope, this means that this arg `is_protected` will equal false as well always, as we cannot have protected cells in non-protected queue and the other way round. I'd remove it completely or add `chassert(!is_protected)`.